### PR TITLE
Add ability to preset AD vector Dirichlet BCs

### LINKED ISSUE #25271: Add ability to preset AD vector Dirichlet BCs
## Reason
This can be valuable for convergence. Additionally it allows equivalent (initial) residual evalutaions for INS and INSAD

## Design
Some templating

## Impact
More flexibility for vector AD Dirichlet boundary conditions

### PR DESCRIPTION
**Pull request by lindsayad**
Newly makes `ADVectorFunctionDirichletBC` preset by default

Closes #25271

### CODE REVIEW

**Comment by GiudGiud** [comment]
if it s preset do we really need AD to contribute to the Jacobian

**Comment by lindsayad** [comment]
Probably not. But with an accurate Jacobian 
- you can be 100% confident that you won't converge to some (wrong) solution other than the trivial solution (delta u = 0)
- you have the option to not preset the boundary condition which can be very useful for instance when displacing a boundary that could potentially lead to inverted elements. If you don't preset in this case, then you allow the solution to move together in the nonlinear iteration to a non-inverted configuration

File: framework/src/bcs/ADDirichletBCBase.C [hunk 1]
```diff
@@ -8,35 +8,20 @@
 //* https://www.gnu.org/licenses/lgpl-2.1.html
 
 #include "ADDirichletBCBase.h"
+#include "MooseVariableFE.h"
+#include "SystemBase.h"
+#include "libmesh/node.h"
```
Comment (GiudGiud) [inline framework/src/bcs/ADDirichletBCBase.C:13]: why do you need more headers with just deleted lines

**Review by GiudGiud — COMMENTED** [review]
looks good.

For templates do we have rules on how to organize the templating parameters?
Like `<A, B>`, which one is the simple one and which one is the complex one? or any other rule?

File: framework/src/bcs/ADDirichletBCBase.C (hunk 1 above)
Comment (lindsayad) [reply framework/src/bcs/ADDirichletBCBase.C:13]: to challenge the compiler 😆

**Comment by lindsayad** [comment]
> For templates do we have rules on how to organize the templating parameters?
> Like `<A, B>`, which one is the simple one and which one is the complex one? or any other rule?

Not really. Sometimes one order just feels more intuitive, like for `DualNumber<T,D>` the first template parameter corresponds to the type of the function value, while the second corresponds to the type of the function derivatives

**Comment by GiudGiud** [comment]
would you say we are in the intuitive case here?
`<T, Base>` with T being the ADornot number and Base being the base class

**Comment by lindsayad** [comment]
I don't know which is more intuitive. As it stands now the first parameter determines whether the class applies to scalar or vector finite element types and the latter parameter essentially decides whether you are going to be the presettable BC or not. If you don't like the ordering just tell me

**Comment by GiudGiud** [comment]
yeah I dont have an opinion

**Review by GiudGiud — APPROVED** [review]
nice consolidation

**Pull request merged**